doc/plugins.dox

```diff
@@ -725,4 +726,402 @@ void geany_load_module(GeanyPlugin *plugin)
+problem) which means the sub-plugin cannot be enabled.
+
+It also installs wrapper functions for the inferior's GeanyPluginFuncs as Ini files aren't code.
+It's very likely that your proxy needs something similar because you can only install function
+pointers to native code.
+
+@code{.c}
+typedef struct {
+	GKeyFile       *file;
+	gchar          *help_text;
+	GSList         *menu_items;
+}
+PluginContext;
+
+
+static gboolean proxy_init(GeanyPlugin *plugin, gpointer pdata);
+static void proxy_help(GeanyPlugin *plugin, gpointer pdata);
+static void proxy_cleanup(GeanyPlugin *plugin, gpointer pdata);
+
+
+static gpointer demoproxy_load(GeanyPlugin *proxy, GeanyPlugin *plugin,
+                               const gchar *filename, gpointer pdata)
+{
+	GKeyFile *file;
+	gboolean result;
+
+	file = g_key_file_new();
+	result = g_key_file_load_from_file(file, filename, 0, NULL);
+
+	if (result)
+	{
+		PluginContext *data = g_new0(PluginContext, 1);
+		data->file = file;
+
+		plugin->info->name = g_key_file_get_locale_string(data->file, "Info", "name", NULL, NULL);
+		plugin->info->description = g_key_file_get_locale_string(data->file, "Info", "description", NULL, NULL);
+		plugin->info->version = g_key_file_get_locale_string(data->file, "Info", "version", NULL, NULL);
+		plugin->info->author = g_key_file_get_locale_string(data->file, "Info", "author", NULL, NULL);
+
+		plugin->funcs->init = proxy_init;
+		plugin->funcs->help = proxy_help;
+		plugin->funcs->cleanup = proxy_cleanup;
+
+		/* Cannot pass g_free as free_func be Geany calls it before unloading, and since 
+		 * demoproxy_unload() accesses the data this would be catastrophic */
+		GEANY_PLUGIN_REGISTER_FULL(plugin, 225, data, NULL);
+		return data;
+	}
+
+	g_key_file_free(file);
+	return NULL;
+}
+@endcode
+
+demoproxy_unload() simply releases all resources aquired in demoproxy_load(). It does not have to
+do anything else in for unloading.
+
+@code{.c}
+static void demoproxy_unload(GeanyPlugin *proxy, GeanyPlugin *plugin, gpointer load_data, gpointer pdata)
+{
+	PluginContext *data = load_data;
+
+	g_free((gchar *)plugin->info->name);
+	g_free((gchar *)plugin->info->description);
+	g_free((gchar *)plugin->info->version);
+	g_free((gchar *)plugin->info->author);
+
+	g_key_file_free(data->file);
+	g_free(data);
+}
+@endcode
+
+Finally the demo_proxy's wrapper GeanyPluginFuncs. They are called for each possible sub-plugin and
+therefore have to multiplex between each using the plugin-defined data pointer. Each is called by
+Geany as if it were an ordinary, native plugin.
+
+proxy_init() actually reads the inferior's file using GKeyFile APIs. It prepares for the help
+dialog and installs the menu items. proxy_help() is called when the user clicks the help button in
+the Plugin Manager. Consequently, this fires up a suitable dialog, although with a dummy message.
+proxy_cleanup() frees all memory allocated in proxy_init().
+
```

Comment:
The entire section is all about the demo proxy. The introductory paragraph tells that the demo proxy loads plugins from ini files for demoing purposes. I don't know how to make it clearer 

Am 25. September 2015 03:43:48 MESZ, schrieb elextr <notifications@github.com>:
>> +	g_free((gchar *)plugin->info->author);
>> +
>> +	g_key_file_free(data->file);
>> +	g_free(data);
>> +}
>> +@endcode
>> +
>> +Finally the demo_proxy's wrapper GeanyPluginFuncs. They are called
>for each possible sub-plugin and
>> +therefore have to multiplex between each using the plugin-defined
>data pointer. Each is called by
>> +Geany as if it were an ordinary, native plugin.
>> +
>> +proxy_init() actually reads the inferior's file using GKeyFile APIs.
>It prepares for the help
>> +dialog and installs the menu items. proxy_help() is called when the
>user clicks the help button in
>> +the Plugin Manager. Consequently, this fires up a suitable dialog,
>although with a dummy message.
>> +proxy_cleanup() frees all memory allocated in proxy_init().
>> +
>
>It isn't clear if the ini file is just something this demo uses to
>define its sub-plugins, or if its required that all proxies use it, and
>if so why?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/geany/geany/pull/629/files#r40393473